benchmark/cublas/blas2/syr2.cpp

```diff
@@ -0,0 +1,188 @@
+  const int v_y_size = 1 + (n - 1) * std::abs(incY);
+
+  // Input matrix/vector, output vector.
+  std::vector<scalar_t> m_a =
+      blas_benchmark::utils::random_data<scalar_t>(n * n);
+  std::vector<scalar_t> v_x =
+      blas_benchmark::utils::random_data<scalar_t>(v_x_size);
+  std::vector<scalar_t> v_y =
+      blas_benchmark::utils::random_data<scalar_t>(v_y_size);
+
+  blas_benchmark::utils::CUDAVector<scalar_t> m_a_gpu(n * n, m_a.data());
+  blas_benchmark::utils::CUDAVector<scalar_t> v_x_gpu(v_x_size, v_x.data());
+  blas_benchmark::utils::CUDAVector<scalar_t> v_y_gpu(v_y_size, v_y.data());
+
+  cublasFillMode_t c_uplo =
+      (*uplo_str == 'u') ? CUBLAS_FILL_MODE_UPPER : CUBLAS_FILL_MODE_LOWER;
+
+#ifdef BLAS_VERIFY_BENCHMARK
+  // Run a first time with a verification of the results
+  std::vector<scalar_t> m_a_ref = m_a;
+  reference_blas::syr2(uplo_str, n, alpha, v_x.data(), incX, v_y.data(), incY,
+                       m_a_ref.data(), lda);
+  std::vector<scalar_t> m_a_temp(m_a);
+
+  {
+    blas_benchmark::utils::CUDAVector<scalar_t, true> m_a_temp_gpu(
+        n * n, m_a_temp.data());
+    cublas_routine<scalar_t>(cuda_handle, c_uplo, n, &alpha, v_x_gpu, incX,
+                             v_y_gpu, incY, m_a_temp_gpu, lda);
+  }
+
+  std::ostringstream err_stream;
+  if (!utils::compare_vectors(m_a_temp, m_a_ref, err_stream, "")) {
+    const std::string& err_str = err_stream.str();
+    state.SkipWithError(err_str.c_str());
+    *success = false;
+  };
+
+#endif
+
+  auto blas_warmup = [&]() -> void {
+    cublas_routine<scalar_t>(cuda_handle, c_uplo, n, &alpha, v_x_gpu, incX,
+                             v_y_gpu, incY, m_a_gpu, lda);
+    return;
+  };
+
+  cudaEvent_t start;
+  cudaEvent_t stop;
+  cudaEventCreate(&start);
+  cudaEventCreate(&stop);
+
+  auto blas_method_def = [&]() -> std::vector<cudaEvent_t> {
+    cudaEventRecord(start);
+    cublas_routine<scalar_t>(cuda_handle, c_uplo, n, &alpha, v_x_gpu, incX,
+                             v_y_gpu, incY, m_a_gpu, lda);
+    cudaEventRecord(stop);
+    cudaEventSynchronize(stop);
+    return std::vector{start, stop};
+  };
+
+  // Warmup
+  blas_benchmark::utils::warmup(blas_method_def);
+  CUDA_CHECK(cudaStreamSynchronize(NULL));
+
+  blas_benchmark::utils::init_counters(state);
+
+  // Measure
+  for (auto _ : state) {
+    // Run
+    std::tuple<double, double> times =
+        blas_benchmark::utils::timef_cuda(blas_method_def);
+
+    // Report
+    blas_benchmark::utils::update_counters(state, times);
+  }
+
+  blas_benchmark::utils::calc_avg_counters(state);
+
+  CUDA_CHECK(cudaEventDestroy(start));
+  CUDA_CHECK(cudaEventDestroy(stop));
+}
+
+template <typename scalar_t>
+void register_benchmark(blas_benchmark::Args& args,
+                        cublasHandle_t* cuda_handle_ptr, bool* success) {
+  // syr2 use the same parameters so reuse spr function
+  auto syr2_params = blas_benchmark::utils::get_syr_params<scalar_t>(args);
```

Comment:
Based on the comment above, shouldn't this be `get_spr_params`?

Reply:
I checked, this is an old comment. Even though `spr` and `syr` share the same parameters for operators call the implementation of `get_spr_params` generates also `incX` and multiple values for `alpha` and in this case I didn't want to benchmark these changes, so I created a new function.
 I will remove the comment, but I think we should try to have a single function that generates parameters for many operators where possible instead of `gen_<operator_name>_params` for each operator.